src/cli.rs

```diff
@@ -949,6 +957,41 @@ impl Pretty {
     }
 }
 
+#[derive(Debug)]
+pub struct FormatOptions {
+    pub json_indent: usize,
+}
+
+impl Default for FormatOptions {
+    fn default() -> Self {
+        Self { json_indent: 4 }
+    }
+}
+
+impl FromStr for FormatOptions {
+    type Err = anyhow::Error;
+    fn from_str(options: &str) -> anyhow::Result<FormatOptions> {
+        let mut format_options = FormatOptions::default();
+
+        // generate a map of the specified options
+        for argument in options.split(',') {
+            let argument_parts: Vec<&str> = argument.splitn(2, ':').collect();
```

Comment:
`split_once` returns an `Option<(&str, &str)>`, thus we'd need to unwrap it first and do some extra error handling. Since we already handle errors in the match statement, I think that'd just be redundant code here.

Reply:
I think with anyhow you can write:
```rust
let (key, value) = argument.split_once(':')
    .context("Format options consist of a key and a value, separated by a \":\".")?;
```
(And then it's easier to have a different message for an unknown key.)

But either is fine as long as it works.

Reply:
@Bnyro Something like this should work:

```rs
match argument.split_once(':') {
    Some(("json.indent", value)) => {
        format_options.json_indent = value.parse().context("Invalid indent value")?;
    }
    _ => {
        return Err(anyhow!(
            "Format options consist of a key and a value, separated by a \":\"."
        ));
    }
}
```